tests/native_api_tests/test_api_model_multi_dimension.cpp

```diff
@@ -0,0 +1,107 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#include "load_dataset.hpp"
+
+#include "power_grid_model_cpp.hpp"
+
+#include <power_grid_model_c/dataset_definitions.h>
+
+#include <doctest/doctest.h>
+
+#include <array>
+#include <complex>
+#include <numbers>
+#include <string>
+#include <vector>
+
+namespace power_grid_model_cpp {
+namespace {
+using namespace std::string_literals;
+using power_grid_model_cpp_test::load_dataset;
+using std::numbers::sqrt3;
+
+// input
+auto const complete_state_json = R"json({
+  "version": "1.0",
+  "type": "input",
+  "is_batch": false,
+  "attributes": {},
+  "data": {
+    "sym_load": [
+      {"id": 2, "node": 0, "status": 1, "type": 0, "p_specified": 0, "q_specified": 0}
+    ],
+    "source": [
+      {"id": 1, "node": 0, "status": 1, "u_ref": 1, "sk": 1e20}
+    ],
+    "node": [
+      {"id": 0, "u_rated": 10e3}
+    ]
+  }
+})json"s;
+
+} // namespace
+
+TEST_CASE("API Model Multi-Dimension") {
+    // model
+    auto const owning_input_dataset = load_dataset(complete_state_json);
+    auto const& input_dataset = owning_input_dataset.dataset;
+    Model model{50.0, input_dataset};
+
+    // 3-D batch update
+    double const u_rated = 10e3;
+    std::vector<double> const u_ref{0.9, 1.0, 1.1};
+    std::vector<double> const p_specified{1e6, 2e6, 3e6, 4e6};
+    std::vector<double> const q_specified{0.1e6, 0.2e6, 0.3e6, 0.4e6, 0.5e6};
+    Idx const size_u_ref = static_cast<Idx>(u_ref.size());
+    Idx const size_p_specified = static_cast<Idx>(p_specified.size());
+    Idx const size_q_specified = static_cast<Idx>(q_specified.size());
```

Comment:
if you use `ssize` then you don't need to `static_cast`

```suggestion
    Idx const size_u_ref = std::ssize(u_ref);
    Idx const size_p_specified = std::ssize(p_specified);
    Idx const size_q_specified = std::ssize(q_specified);
```
or even
```suggestion
    auto const size_u_ref = std::ssize(u_ref);
    auto const size_p_specified = std::ssize(p_specified);
    auto const size_q_specified = std::ssize(q_specified);
```